core/common/src/main/java/alluxio/concurrent/ManagedBlockingUfsForwarder.java

```diff
@@ -0,0 +1,654 @@
+      }
+    }.get();
+  }
+
+  @Override
+  public boolean exists(String path) throws IOException {
+    return new ManagedBlockingUfsMethod<Boolean>() {
+      @Override
+      public Boolean execute() throws IOException {
+        return mUfs.exists(path);
+      }
+    }.get();
+  }
+
+  @Override
+  public Pair<AccessControlList, DefaultAccessControlList> getAclPair(String path)
+      throws IOException {
+    return new ManagedBlockingUfsMethod<Pair<AccessControlList, DefaultAccessControlList>>() {
+      @Override
+      public Pair<AccessControlList, DefaultAccessControlList> execute() throws IOException {
+        return mUfs.getAclPair(path);
+      }
+    }.get();
+  }
+
+  @Override
+  public long getBlockSizeByte(String path) throws IOException {
+    return new ManagedBlockingUfsMethod<Long>() {
+      @Override
+      public Long execute() throws IOException {
+        return mUfs.getBlockSizeByte(path);
+      }
+    }.get();
+  }
+
+  @Override
+  public UfsDirectoryStatus getDirectoryStatus(String path) throws IOException {
+    return new ManagedBlockingUfsMethod<UfsDirectoryStatus>() {
+      @Override
+      public UfsDirectoryStatus execute() throws IOException {
+        return mUfs.getDirectoryStatus(path);
+      }
+    }.get();
+  }
+
+  @Override
+  public UfsDirectoryStatus getExistingDirectoryStatus(String path) throws IOException {
+    return new ManagedBlockingUfsMethod<UfsDirectoryStatus>() {
+      @Override
+      public UfsDirectoryStatus execute() throws IOException {
+        return mUfs.getExistingDirectoryStatus(path);
+      }
+    }.get();
+  }
+
+  @Override
+  public List<String> getFileLocations(String path) throws IOException {
+    return new ManagedBlockingUfsMethod<List<String>>() {
+      @Override
+      public List<String> execute() throws IOException {
+        return mUfs.getFileLocations(path);
+      }
+    }.get();
+  }
+
+  @Override
+  public List<String> getFileLocations(String path, FileLocationOptions options)
+      throws IOException {
+    return new ManagedBlockingUfsMethod<List<String>>() {
+      @Override
+      public List<String> execute() throws IOException {
+        return mUfs.getFileLocations(path, options);
+      }
+    }.get();
+  }
+
+  @Override
+  public UfsFileStatus getFileStatus(String path) throws IOException {
+    return new ManagedBlockingUfsMethod<UfsFileStatus>() {
+      @Override
+      public UfsFileStatus execute() throws IOException {
+        return mUfs.getFileStatus(path);
+      }
+    }.get();
+  }
+
+  @Override
+  public UfsFileStatus getExistingFileStatus(String path) throws IOException {
+    return new ManagedBlockingUfsMethod<UfsFileStatus>() {
+      @Override
+      public UfsFileStatus execute() throws IOException {
+        return mUfs.getExistingFileStatus(path);
+      }
+    }.get();
+  }
+
+  @Override
+  public String getFingerprint(String path) {
+    return mUfs.getFingerprint(path);
```

Comment:
Should this be wrapped?

Reply:
Does getting a fingerprint entail an UFS interaction? If it does, is it slow?

Reply:
It varies based on the implementation, some call `getStatus` which will be slow (but that is already wrapped in ManagedBlocker).

Reply:
Then it's better to exclude it and postpone managed blocking only when `getStatus` is called.